sdcm/provision/common/provision_plan.py

```diff
@@ -0,0 +1,34 @@
+import logging
+from typing import List
+
+from pydantic import BaseModel
+
+from sdcm.provision.common.provisioner import ProvisionParameters, InstanceProvisionerBase, InstanceParamsBase, TagsType
+
+LOGGER = logging.getLogger(__name__)
+
+
+class ProvisionPlan(BaseModel):
+    provision_steps: List[ProvisionParameters]
+    provisioner: InstanceProvisionerBase
+
+    @property
+    def name(self):
+        return self.__class__.__name__
+
+    def provision_instances(self, instance_parameters: InstanceParamsBase, node_count: int, node_tags: List[TagsType],
+                            node_names: List[str]):
+        for provision_parameters in self.provision_steps:
+            if instances := self.provisioner.provision(
+                    provision_parameters=provision_parameters,
+                    instance_parameters=instance_parameters,
+                    count=node_count,
+                    tags=node_tags,
+                    names=node_names,
+            ):
+                LOGGER.info('%s: Instances has been provisioned using "%s":\n%s', self.name,
+                            provision_parameters.name, instances)
+                return instances
+            else:
+                LOGGER.error('%s: Failed has been provisioned using "%s"', self.name, provision_parameters.name)
```

Comment:
'Failed has been provisioned using "%s"' sounds strange.
Maybe 'Failed to provision instances using "%s"'"?

Reply:
fixed